src/test/scala/com/databricks/spark/avro/AvroSuite.scala

```diff
@@ -21,6 +22,20 @@ class AvroSuite extends FunSuite {
   val episodesFile = "src/test/resources/episodes.avro"
   val testFile = "src/test/resources/test.avro"
 
+  test("reading and writing partitioned data") {
+    TestUtils.withTempDir { dir =>
```

Comment:
How about moving the `withTempDir` call into the `for` loop? So that you don't need the UUID trick to create unique directories.